File: R/cma_score.R

compute_cma <- function(y, x){
  x_rank <- rank(x, ties.method = "average")
  y_rank <- rank(y, ties.method = "average")
  return(0.5*(cov(y_rank, x_rank) / cov(y_rank, y_rank) + 1))
}

File: Python/cma_score.py
import numpy as np
from scipy.stats import rankdata

def compute_cma(y, x):
    y_rank = rankdata(y, method='average')
    x_rank = rankdata(x, method='average') 
    N = len(y)
    #mean_rank = (N+1)/2
    var = np.sum((y_rank - np.mean(y_rank))**2)*(1/(N-1))
    return (np.cov(y_rank,x_rank)[0][1]/var+1)/2